Automattic/newspack-plugin
# Feat/stripe and wc subscriptions

### All Submissions:






### Changes proposed in this Pull Request:



Adds basic synchronisation with WooCommerce Subscriptions. The Stripe subscriptions will be synchronised to WC Subscriptions, but won't yet be editable (cancellable) _from_ WooCommerce Subscriptions (will be in another PR). 

TODO: 

- [ ] subscription cancellation
- [ ] subscription reactivation
- [ ] subscription pausing 

### How to test the changes in this Pull Request:

1. Start with a site with Reader Activation configured, Stripe set as Reader Revenue Platform, and the WooCommerce suite (including WooCommerce Subscriptions) active. Ensure Stripe can reach your test site with a webhook call.
2. Create a new reader, e.g. by registering on the front-end
3. We'll use [Stripe's test clocks](https://stripe.com/docs/billing/testing/test-clocks) to travel in time - visit the [test clocks in Stripe dashboard](https://dashboard.stripe.com/test/test-clocks) and create a new one
4. Add a new customer in the test clock, and add a payment method to them
5. Edit the customer metadata and add the new reader WP user ID as `userId` metadata*
6. Still in Stripe's test clock, add a new monthly-donation-subscription to the customer
7. Visit WP Admin -> WooCommerce Subscriptions and observe a new Subscription has been created
8. _Advance time_ on the Stripe test clock by a month
9. Observe a subscription renewal happens – a new charge is made by Stripe, and a new order add in WooCommerce
10. Observe the order is associated with the WC Subscription

\* This is all because test-clock-customers can't be created as a result of the donation, as usual customers are. The test-clock-customers are on a different time plane and thus have to be created manually and linked to the Newspack reader

### Other information:
